tests/deepspeed/test_deepspeed.py

```diff
@@ -176,27 +175,26 @@ def test_init_zero3_fp16(self):
             },
         }
 
-        dschf = HfDeepSpeedConfig(ds_config)
-
-        self.assertTrue(dschf.is_zero3())
-        self.assertTrue(is_deepspeed_zero3_enabled())
-
         with LoggingLevel(logging.INFO):
             with mockenv_context(**self.dist_env_1_gpu):
+                dschf = HfTrainerDeepSpeedConfig(ds_config)
+                AcceleratorState.set_deepspeed_config(dschf)
+                self.assertTrue(dschf.is_zero3())
+                self.assertTrue(is_deepspeed_zero3_enabled())
                 logger = logging.get_logger("transformers.modeling_utils")
                 with CaptureLogger(logger) as cl:
                     AutoModel.from_pretrained(T5_TINY)
         self.assertIn("Detected DeepSpeed ZeRO-3", cl.out)
 
-        # now remove zero optimization
-        del ds_config["zero_optimization"]
-        dschf = HfDeepSpeedConfig(ds_config)
-
-        self.assertFalse(dschf.is_zero3())
-        self.assertFalse(is_deepspeed_zero3_enabled())
```

Comment:
please revert - stage=0 is a thing (used by new bf16 optim)

Reply:
I think the only thing this PR needs to touch is deepspeed.py to create wrappers to the code that moved, everything else should remain as it was before to keep things BC.